File: src/main/java/com/example/stockfx/entities/Fournisseur.java
package com.example.stockfx.entities;

public class Fournisseur {

    private int idFournisseur;
    private String nom_fournisseur;
    private String num_telephone;
    private String num_fixe;
    private String email;
    private String matricule;
    private String type_véhicule;
    private String photo;

    public Fournisseur(){

    }
    public Fournisseur(String nom_fournisseur ,String num_telephone, String num_fixe, String email, String matricule, String photo) {
        this.nom_fournisseur = nom_fournisseur;
        this.num_telephone = num_telephone;
        this.num_fixe = num_fixe;
        this.email = email;
        this.matricule = matricule;
        this.photo = photo;
    }
    public Fournisseur(int id_fournisseur,String nom_fournisseur, String num_telephone, String num_fixe, String email, String matricule, String photo) {
        this.idFournisseur=id_fournisseur;
        this.nom_fournisseur = nom_fournisseur;
        this.num_telephone = num_telephone;
        this.num_fixe = num_fixe;
        this.email = email;
        this.matricule = matricule;
        this.type_véhicule = type_véhicule;
        this.photo = photo;
    }



    public String getNom_fournisseur() {
        return nom_fournisseur;
    }

    public void setNom_fournisseur(String nom_fournisseur) {
        this.nom_fournisseur = nom_fournisseur;
    }

    public String getNum_telephone() {
        return num_telephone;
    }

    public void setNum_telephone(String num_telephone) {
        this.num_telephone = num_telephone;
    }

    public String getNum_fixe() {
        return num_fixe;
    }

    public void setNum_fixe(String num_fixe) {
        this.num_fixe = num_fixe;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getMatricule() {
        return matricule;
    }

    public void setMatricule(String matricule) {
        this.matricule = matricule;
    }

    public String getType_véhicule() {
        return type_véhicule;
    }

    public void setType_véhicule(String type_véhicule) {
        this.type_véhicule = type_véhicule;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }



    public int getId_fournisseur() {
        return idFournisseur;
    }

    public void setId_fournisseur(int idFournisseur) {
        this.idFournisseur = idFournisseur;
    }
    @Override
    public String toString() {
        return "Fournisseur{" +
                "idFournisseur=" + idFournisseur +
                ", nom_fournisseur='" + nom_fournisseur + '\'' +
                ", num_telephone='" + num_telephone + '\'' +
                ", num_fixe='" + num_fixe + '\'' +
                ", email='" + email + '\'' +
                ", matricule='" + matricule + '\'' +
                ", type_véhicule='" + type_véhicule + '\'' +
                ", photo='" + photo + '\'' +
                '}';
    }



}

File: src/Entity/Fournisseur.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\HttpFoundation\File\File;
use Vich\UploaderBundle\Mapping\Annotation as Vich;

use Vich\UploaderBundle\Entity\File as EmbeddedFile;

/**
 * Fournisseur
 *
 * @ORM\Table(name="fournisseur")
 * @ORM\Entity
 * @Vich\Uploadable
 */
class Fournisseur
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_fournisseur", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idFournisseur;

    /**
     * @var string
     *
     * @ORM\Column(name="nom_fournisseur", type="string", length=30, nullable=false)
     */
    private $nomFournisseur;

    /**
     * @var string
     *
     * @ORM\Column(name="numtel_fournisseur", type="string", length=20, nullable=false)
     */
    private $numtelFournisseur;

    /**
     * @var string
     *
     * @ORM\Column(name="numFixe_fournisseur", type="string", length=20, nullable=false)
     */
    private $numfixeFournisseur;

    /**
     * @var string
     *
     * @ORM\Column(name="email", type="string", length=255, nullable=false)
     */
    private $email;

    /**
     * @var string
     *
     * @ORM\Column(name="matricule_fiscale", type="string", length=20, nullable=false)
     */
    private $matriculeFiscale;

    /**
     * @var string
     *
     * @ORM\Column(name="photo", type="string", length=255, nullable=false)
     */
    private $photo;

    /**
     * @Vich\UploadableField(mapping="fournisseurs",fileNameProperty="photo")
     * @var File|null
     */
    private $imageFile;

    /**
     * @return File
     */
    public function getImageFile(): ?File
    {
        return $this->imageFile;
    }

    /**
     * @param File $imageFile
     */
    public function setImageFile(?File $imageFile = null): void    {
        $this->imageFile = $imageFile;
    }
    public function setIdFournisseur(int $IdFournisseur): self
    {
        $this->IdFournisseur = $IdFournisseur;

        return $this;
    }
    public function getIdFournisseur(): ?int
    {
        return $this->idFournisseur;
    }

    public function getNomFournisseur(): ?string
    {
        return $this->nomFournisseur;
    }

    public function setNomFournisseur(string $nomFournisseur): self
    {
        $this->nomFournisseur = $nomFournisseur;

        return $this;
    }

    public function getNumtelFournisseur(): ?string
    {
        return $this->numtelFournisseur;
    }

    public function setNumtelFournisseur(string $numtelFournisseur): self
    {
        $this->numtelFournisseur = $numtelFournisseur;

        return $this;
    }

    public function getNumfixeFournisseur(): ?string
    {
        return $this->numfixeFournisseur;
    }

    public function setNumfixeFournisseur(string $numfixeFournisseur): self
    {
        $this->numfixeFournisseur = $numfixeFournisseur;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }

    public function getMatriculeFiscale(): ?string
    {
        return $this->matriculeFiscale;
    }

    public function setMatriculeFiscale(string $matriculeFiscale): self
    {
        $this->matriculeFiscale = $matriculeFiscale;

        return $this;
    }

    public function getPhoto(): ?string
    {
        return $this->photo;
    }

    public function setPhoto(string $photo):void
    {
        $this->photo = $photo;

        //return $this;
    }
    public function __toString()
    {
        return $this->nomFournisseur;
    }



}
